encoding/marshal_test.go

```diff
@@ -216,18 +216,58 @@ func TestEncodeDecode(t *testing.T) {
 	}
 }
 
+func TestEncodeAll(t *testing.T) {
+	var expected []byte
+	for i := range testStructs {
+		expected = append(expected, Marshal(testStructs[i])...)
+	}
+
+	b := new(bytes.Buffer)
+	NewEncoder(b).EncodeAll(testStructs...)
+	if !bytes.Equal(b.Bytes(), expected) {
+		t.Errorf("expected %v, got %v", expected, b.Bytes())
+	}
+}
+
+func TestDecodeAll(t *testing.T) {
+	b := new(bytes.Buffer)
+	enc := NewEncoder(b)
+	for i := range testStructs {
+		enc.Encode(testStructs[i])
+	}
```

Comment:
A similar thing here. Instead of (or rather in addition to) encoding something and then decoding it, I think it makes sense to start with some hardcoded bytes and some data and verify that it decodes correctly.